api/src/certification/evaluation/domain/services/scoring/calibrated-challenge-service.js

```diff
@@ -1,49 +1,50 @@
+ * @typedef {import('../index.js').CalibratedChallengeRepository} CalibratedChallengeRepository
+ * @typedef {import('../../../../../shared/domain/models/Challenge.js').Challenge} Challenge
+ * @typedef {import('../../../../scoring/domain/read-models/ChallengeCalibration.js').ChallengeCalibration} ChallengeCalibration
+ * @typedef {import('../../../../shared/domain/models/Version.js').Version} Version
+ * @typedef {import('../../../../shared/domain/models/CertificationCourse.js').CertificationCourse} CertificationCourse
+ * @typedef {import('../../models/CalibratedChallenge.js').CalibratedChallenge} CalibratedChallenge
+ */
+/**
+ * @typedef {Object} FindByCertificationCourseIdAndAssessmentIdResult
+ * @property {Array<CalibratedChallenge>} allChallenges - All calibrated challenges for the version no matter the locale
+ * @property {Array<CalibratedChallenge>} askedChallengesWithoutLiveAlerts - Calibrated challenges presented to the candidate, excluding those with validated live alerts.
+ * @property {Array<ChallengeCalibration>} challengeCalibrationsWithoutLiveAlerts - Calibrations of challenges presented to the candidate, excluding those with validated live alerts.
  */
-import differenceBy from 'lodash/differenceBy.js';
-
-import { config } from '../../../../../shared/config.js';
 import { withTransaction } from '../../../../../shared/domain/DomainTransaction.js';
 
 export const findByCertificationCourseIdAndAssessmentId = withTransaction(
   /**
    * @param {Object} params
+   * @param {CertificationCourse} params.certificationCourse
+   * @param {Version} params.version
    * @param {ChallengeCalibrationRepository} params.challengeCalibrationRepository
    * @param {CertificationChallengeLiveAlertRepository} params.certificationChallengeLiveAlertRepository
-   * @param {SharedChallengeRepository} params.sharedChallengeRepository
-   * @param {ChallengeRepository} params.challengeRepository
-   * @param {CertificationCourseRepository} params.certificationCourseRepository
+   * @param {CalibratedChallengeRepository} params.calibratedChallengeRepository
+   * @returns {Promise<FindByCertificationCourseIdAndAssessmentIdResult>}
    */
   async ({
-    certificationCourseId,
-    assessmentId,
+    certificationCourse,
+    version,
     challengeCalibrationRepository,
     certificationChallengeLiveAlertRepository,
-    sharedChallengeRepository,
-    challengeRepository,
-    certificationCourseRepository,
+    calibratedChallengeRepository,
   }) => {
-    const fromArchivedCalibration = await _isOldCalibration({ certificationCourseId, certificationCourseRepository });
-
-    const flashCompatibleChallenges = await challengeRepository.findFlashCompatibleWithoutLocale({
-      useObsoleteChallenges: true,
-      fromArchivedCalibration,
-    });
+    const flashCompatibleChallenges = await calibratedChallengeRepository.findAllCalibratedChallenges({ version });
```

Comment:
```suggestion
    const versionCalibratedChallenges = await calibratedChallengeRepository.findAllCalibratedChallenges({ version });
```

Est-ce que c'est toujours utile de parler de `flashCompatible` ici ?

Reply:
Nom modifie, done